api/v1alpha1/apidefinition_types.go

```diff
@@ -699,12 +720,26 @@ type BasicAuthMeta struct {
 	BodyPasswordRegexp string `json:"body_password_regexp"`
 }
 
+// +
 type OAuth2Meta struct {
-	AllowedAccessTypes     []string `json:"allowed_access_types"`    // osin.AccessRequestType
-	AllowedAuthorizeTypes  []string `json:"allowed_authorize_types"` // osin.AuthorizeRequestType
-	AuthorizeLoginRedirect string   `json:"auth_login_redirect"`
+
+	// AllowedAccessTypes are an array of allowable access types.
+	AllowedAccessTypes []AccessTypeEnum `json:"allowed_access_types"` // osin.AccessRequestType
+
+	// AllowedAuthorizeTypes is an array of allowable `response_type` parameters `token` or authorization code `code`.
+	// Choose token for client_credentials or implicit grant types.
+	AllowedAuthorizeTypes []AuthorizeTypeEnum `json:"allowed_authorize_types"` // osin.AuthorizeRequestType
+
+	// Login form to handle user login.
+	AuthLoginRedirect string `json:"auth_login_redirect,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=authorization_code;refresh_token;password;client_credentials
```

Comment:
according to osin, `__implicit` and `assertion` are missing

```suggestion
// +kubebuilder:validation:Enum=authorization_code;refresh_token;password;client_credentials;assertion;__implicit
```

Reply:
<img width="389" alt="Screenshot 2021-03-29 at 10 34 20" src="https://user-images.githubusercontent.com/1465130/112817434-56b45200-907a-11eb-8268-3aaa77837a7a.png">

Thanks for the suggestion. According to Tyk Docs - we don't support implicit grant. And i don't know what `assertion` is. I also am not so keen on users relying on undocumented features. So would prefer not to include for the time being.